### PR TITLE
fix positional only args

### PR DESCRIPTION
**Pull request by MyGodIsHe**

### CODE REVIEW

**Comment by jacksongoode** [comment]
What case does this resolve? Is it only an issue < 3.8?

**Comment by MyGodIsHe** [comment]
This error occurs in python 3.8 and higher. There are problematic cases in the tests - func2 and func3.

File: micro-benchmark/snippets/args/positional_only_call/main.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    pass
+
+
+def func3(q, a=1, b=2, /, c=3):
+    pass
+
+
+func1(0)
+func2()
+func3(0)
```
Comment (vitsalis) [inline micro-benchmark/snippets/args/positional_only_call/main.py:15]: Can those tests also contain the usage of a keyword argument?

File: pycg/processing/preprocessor.py [hunk 2]
```diff
@@ -50,8 +50,18 @@ def __init__(
 
     def _get_fun_defaults(self, node):
         defaults = {}
-        start = len(node.args.args) - len(node.args.defaults)
-        for cnt, d in enumerate(node.args.defaults, start=start):
+        posonlyargs = getattr(node.args, 'posonlyargs', [])  # for python version less than 3.8
+        start = len(posonlyargs) + len(node.args.args) - len(node.args.defaults)
+        defaults_sep = len(posonlyargs) - start
```
Comment (vitsalis) [inline pycg/processing/preprocessor.py:55]: Doesn't this imply that `defaults_sep = len(node.args.defaults) - len(node.args.args)`? If so, then wouldn't this imply that `start` is negative?

**Review by vitsalis — COMMENTED** [review]
Thanks for the contribution and apologies for the late response. Unfortunately I do not have much available time for the maintenance for this repository.

The logic added on the `_get_fun_defaults` is a bit confusing for me. Added a couple of comments regarding it. I would appreciate if you added comments explaining what exactly is happening in the code and also adding an example.

**Comment by vitsalis** [comment]
Closing due to archival of repository.

**Pull request closed without merging**